crates/debugger/src/tui/draw.rs

```diff
@@ -350,7 +352,20 @@ impl DebuggerContext<'_> {
                 let pc_ic_map = if is_create { create_map } else { rt_map };
                 let ic = pc_ic_map.get(pc)?;
                 let source_element = source_map.swap_remove(ic);
```

Comment:
I have a suspicion that if two contracts share the same name, this could retrieve a `source_element` that doesn't actually belong to the contract being called – but fixing that case probably involves a lot of upstream changes. Just noting here.

Reply:
that might be right